### PR TITLE
add test for `status`=`error`

### PR DESCRIPTION
**Pull request by KATT**
Related to https://github.com/trpc/trpc/pull/1645

### CODE REVIEW

File: src/react/tests/useQuery.test.tsx [hunk 1]
```diff
@@ -4677,4 +4677,38 @@ describe('useQuery', () => {
...
+    const error = new Error('oops')
+
+    const queryFn = async () => {
+      throw error
+    }
+
+    function Page() {
+      const state = useQuery(key, queryFn, {
+        retry: false,
+        retryOnMount: false,
```
Comment (KATT) [inline src/react/tests/useQuery.test.tsx:4694]: Cc @TkDodo what puzzles me here is that the `retryOnMount: false` is needed. Shouldn't `retry: false` override it?

Comment (TkDodo) [reply src/react/tests/useQuery.test.tsx:4694]: it's a different setting. `retryOnMount` might be not ideally named:
-  `retry` retries internally when a query fails, and then it goes to `error` state after all retries are done.
- `retryOnMount` is more like `refetchOnMountIfWeHaveAnErrorButNoDataYet`

**Pull request merged**